app/commands/create_grade_year.rb

```diff
@@ -0,0 +1,25 @@
+class CreateGradeYear < ROM::Commands::Create[:sql]
+  relation :grade_years
+  register_as :create
+  result :one
+
+  def execute(attributes)
+    gpa = attributes.delete(:gpa).to_f
+    if gpa > 3
+      letter = 'A'
+    elsif gpa > 1
+      letter = 'D'
+    end
+
+    grade = Grade.find_by_letter(letter)
```

Comment:
Something else should take care of this. I'd imagine some `CreateGrade` object that would fetch grade by letter inferred from gpa and just pass it to the command.